packages/core/src/dynamoDb/utils/dynamodb.ts

```diff
@@ -0,0 +1,80 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DynamoDB } from 'aws-sdk'
+import { AttributeValue, Key, ScanInput } from 'aws-sdk/clients/dynamodb'
+import { DynamoDbClient } from '../../shared/clients/dynamoDbClient'
+
+export interface RowData {
+    [key: string]: string
+}
+export interface TableData {
+    tableHeader: RowData[]
+    tableContent: RowData[]
+    lastEvaluatedKey?: Key
+}
+
+export async function getTableContent(
+    tableRequest: ScanInput,
+    regionCode: string,
+    client = new DynamoDbClient(regionCode)
+) {
+    const response = await client.scanTable(tableRequest)
+    const tableColumnsNames = getTableColumnsNames(response)
+    const tableItems = getTableItems(tableColumnsNames[0], response)
+
+    const tableData: TableData = {
+        tableHeader: tableColumnsNames[1],
+        tableContent: tableItems,
+        lastEvaluatedKey: response.LastEvaluatedKey,
+    }
+    return tableData
+}
+
+export function getTableColumnsNames(items: DynamoDB.Types.ScanOutput): [Set<string>, RowData[]] {
+    const tableColumnsNames = new Set<string>()
+    const tableHeader = [] as RowData[]
+    for (const item of items.Items ?? []) {
+        for (const key of Object.keys(item)) {
+            tableColumnsNames.add(key)
+        }
+    }
+    for (const columnName of tableColumnsNames) {
+        tableHeader.push({ columnDataKey: columnName, title: columnName })
+    }
+
+    return [tableColumnsNames, tableHeader]
```

Comment:
rather than returning a set and indexing it as an array does it make sense to just return:
```
{
    columns: tableColumnsNames,
    header: tableHeader
}
```
? Typescript is much more flexible then other languages on this

Reply:
Updated